editor/scene_tree_dock.cpp

```diff
@@ -3287,6 +3338,9 @@ void SceneTreeDock::_update_tree_menu() {
 	tree_menu->add_check_item(TTR("Auto Expand to Selected"), TOOL_AUTO_EXPAND);
 	tree_menu->set_item_checked(tree_menu->get_item_index(TOOL_AUTO_EXPAND), EDITOR_GET("docks/scene_tree/auto_expand_to_selected"));
 
+	tree_menu->add_check_item(TTR("Center Node on Reparent"), TOOL_CENTER_PARENT);
```

Comment:
I think this needs a tooltip explaining what it does, as it's not obvious.
Something like: `If enabled, Reparent to New Node option will create the new node in the center of the selected nodes, if  possible.`